core/src/main/java/org/p2p/core/token/Token.kt

```diff
@@ -188,3 +188,7 @@ sealed class Token constructor(
         }
     }
 }
+
+fun List<Token.Active>.findSol(): Token.Active = first { it.isSOL }
```

Comment:
`requireSol` would be better to show that it can throw exception

Reply:
good idea! but it's just doesn't look semantic right

tokens.requireSol() looks like just a check
i made a contract based on List operations like firstOrNull first() that we are all know

require, requireNotNull from kotlin checks have return type `Nothing`

Reply:
Okay. 
You may add `throw` maybe, smth like `findSolOrThrow` but it's very optional.